agent/api/task.go

```diff
@@ -1164,3 +1169,66 @@ func (task *Task) RecordExecutionStoppedAt(container *Container) {
 	seelog.Infof("Task [%s]: recording execution stopped time. Essential container [%s] stopped at: %s",
 		task.Arn, container.Name, now.String())
 }
+
+// RequiresASMDockerAuthData returns true if atleast one container in the task
+// needs to retrieve private registry authentication data from ASM
+func (task *Task) RequiresASMDockerAuthData() bool {
+	for _, container := range task.Containers {
+		if container.ShouldPullWithASMAuth() {
+			return true
+		}
+	}
+	return false
+}
+
+// RetrieveASMDockerAuthData uses the execution role to retrieve private
+// registry authentication data from ASM
+func (task *Task) RetrieveASMDockerAuthData(asmAuthData *ASMAuthData, credentialsManager credentials.Manager) (docker.AuthConfiguration, error) {
+	executionCredentials, ok := credentialsManager.GetTaskCredentials(task.GetExecutionCredentialsID())
```

Comment:
so this feature is tied to the `executionRole` feature?

Reply:
yes it is.